host/entry.go

```diff
@@ -0,0 +1,293 @@
+// device connectivity.
+type DevChange struct {
+	Sense DevChangeSense
+	Dev   *libsio.Dev
+}
+
+// Capture tries to open an input such as a microphone.
+func Capture(e Entry) (sound.Source, error) {
+	return CaptureWith(e, e.DefaultForm(), e.DefaultSampleCodec(), e.DefaultBufSize())
+}
+
+// CaptureWith opens audio capture such as via a microphone
+// with the specified sound.Form v, sample.Codec co, and buffer size b.
+func CaptureWith(e Entry, v sound.Form, co sample.Codec, b int) (sound.Source, error) {
+	if !e.CanOpenSource() {
+		return nil, ErrUnsupported
+	}
+	dev := e.DefaultInputDev()
+	s, _, err := e.OpenSource(dev, v, co, b)
+	return s, err
+}
+
+// Play plays a sound.Source
+func Play(e Entry, src sound.Source) error {
+	snk, err := Player(e, src)
+	if err != nil {
+		return err
+	}
+	return ops.Copy(snk, src)
+}
+
+// PlayWith returns a Player for src with output sample codec co
+// and buffer size b.
+func PlayWith(e Entry, src sound.Source, co sample.Codec, b int) error {
+	snk, err := PlayerWith(e, src, co, b)
+	if err != nil {
+		return err
+	}
+	return ops.Copy(snk, src)
+}
+
+// Player tries to return a sound.Sink to which
+// writes are played.
+func Player(e Entry, v sound.Form) (sound.Sink, error) {
+	return PlayerWith(e, v, e.DefaultSampleCodec(), e.DefaultBufSize())
+}
+
+// PlayerWith opens a sound.Sink for playback with the specified sound.Form,
+// sample.Codec, and buffersize.
+func PlayerWith(e Entry, v sound.Form, co sample.Codec, b int) (sound.Sink, error) {
+	if !e.CanOpenSink() {
+		return nil, ErrUnsupported
+	}
+	dev := e.DefaultOutputDev()
+	snk, _, err := e.OpenSink(dev, v, co, b)
+	if err != nil {
+		return nil, err
+	}
+	return snk, nil
+}
+
+// Duplex returns a duplex with entry defaults.
+func Duplex(e Entry, iv, ov sound.Form) (sound.Duplex, error) {
+	return DuplexWith(e, iv, ov, e.DefaultSampleCodec(), e.DefaultBufSize())
+}
+
+// DuplexWith opens a Duplex connection with the specified input, output forms,
+// sample codec, and buffer size.
+func DuplexWith(e Entry, iv, ov sound.Form, co sample.Codec, b int) (sound.Duplex, error) {
+	if !e.CanOpenDuplex() {
+		return nil, ErrUnsupported
+	}
+	dev := e.DefaultDuplexDev()
+	dpx, _, err := e.OpenDuplex(dev, iv, ov, co, b)
+	return dpx, err
+}
+
+type entry struct {
+	Entry
+	pkgPath string
+}
+
+func pkgPath(v interface{}) string {
+	typ := reflect.ValueOf(v).Type()
+	return typ.PkgPath()
+}
+
+var entries map[string][]*entry = make(map[string][]*entry)
+
+// RegisterEntry registers an Entry
+func RegisterEntry(e Entry) error {
+	nm := e.Name()
+	found := false
+	for _, okName := range Names() {
+		if nm == okName {
+			found = true
+			break
+		}
+	}
+	if !found {
+		return ErrInvalidEntryName
+	}
+	privEntry := &entry{
+		Entry:   e,
+		pkgPath: pkgPath(e)}
+	entries[nm] = append(entries[nm], privEntry)
+	return nil
+}
+
+var hMu sync.Mutex
+var eMu sync.Mutex
+var theEntry Entry
+
+// Get opens the default entry for the host.
```

Comment:
Fix docs or function name. Which of `Get` and `Connect` is correct?

Reply:
Connect is "correct" in that it is the best name I came up with so far.